File: frontend/src/objects/order/core/OrderSummaryActivity.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type { OrderPartialRefundRequest } from '@/objects/order/apiTypes/OrderPartialRefundRequest'
import type { OrderChatMessage } from '@/objects/order/core/OrderChatMessage'
import type { OrderTimelineEntry } from '@/objects/order/core/OrderTimelineEntry'

export type OrderSummaryActivity = {
  timeline: OrderTimelineEntry[]
  chatMessages: OrderChatMessage[]
  partialRefundRequests: OrderPartialRefundRequest[]
}

File: backend/src/main/scala/services/order/objects/OrderSummaryActivity.scala
package services.order.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import services.order.objects.apiTypes.*

final case class OrderSummaryActivity(
    timeline: List[OrderTimelineEntry],
    chatMessages: List[OrderChatMessage],
    partialRefundRequests: List[OrderPartialRefundRequest],
)
